app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryListViewModel.kt

```diff
@@ -8,36 +8,61 @@ import androidx.lifecycle.ViewModel
 import org.oppia.android.R
 import org.oppia.android.app.home.HomeItemViewModel
 import org.oppia.android.app.home.RouteToRecentlyPlayedListener
+import org.oppia.android.app.model.RecommendedActivityList
 
 /** [ViewModel] for the promoted story list displayed in [HomeFragment]. */
 class PromotedStoryListViewModel(
   private val activity: AppCompatActivity,
-  val promotedStoryList: List<PromotedStoryViewModel>
+  val promotedStoryList: List<PromotedStoryViewModel>,
+  private val recommendedActivityList: RecommendedActivityList
 ) : HomeItemViewModel() {
   private val routeToRecentlyPlayedListener = activity as RouteToRecentlyPlayedListener
 
-  // TODO(#2297): Update this span count and move to values/integers.xml once behavior is clarified
-  private val promotedStoriesTabletSpanCount: Int =
-    if (Resources.getSystem().configuration.orientation == Configuration.ORIENTATION_PORTRAIT) 2
-    else 3
-
   /** Returns the padding placed at the end of the promoted stories list based on the number of promoted stories. */
   val endPadding =
     if (promotedStoryList.size > 1)
       activity.resources.getDimensionPixelSize(R.dimen.home_padding_end)
     else activity.resources.getDimensionPixelSize(R.dimen.home_padding_start)
 
   /** Determines and returns the visibility for the "View All" button. */
+  fun getHeader(): String {
+    recommendedActivityList.recommendedStoryList.let {
+      return when {
+        it.suggestedStoryCount != 0 -> {
+          if (it.recentlyPlayedStoryCount != 0 || it.olderPlayedStoryCount != 0) {
+            activity.getString(R.string.stories_for_you)
+          } else
+            activity.getString(R.string.recommended_stories)
+        }
+        it.recentlyPlayedStoryCount != 0 -> {
```

Comment:
how about if we go with `> 0` this way the value can never go in negative if in case the value messed up?

Reply:
Yes, done.